### ISSUE TITLE
image_dataset_from_directory supporting reshuffle_each_iteration

### ISSUE THREAD

**Issue by jamesbraza** [keras-team/keras#17264]
**System information**.

TensorFlow version (you are using): 20.10.0
Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state**.

I find [`tf.keras.utils.image_dataset_from_directory`](https://www.tensorflow.org/api_docs/python/tf/keras/utils/image_dataset_from_directory) very convenient for reading in images from data sources like Kaggle.  I use `shuffle=True` so that training subset sees a distribution of all classes.

The [`Dataset.shuffle`](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#shuffle) API supports an arg `reshuffle_each_iteration` that makes each `Dataset` iteration have the same shuffled ordering.  This `reshuffle_each_iteration = False` behavior currently is not supported by `image_dataset_from_directory`.

I am trying to compare the training process of several `Model`s.  Having the same ordering of shuffled images is valuable to me, for comparability.  Within one Python script, I would like to:
1. Read in a dataset using `image_dataset_from_directory`
2. Pass the same ordering of shuffled examples to 2+ `model.fit` calls

If `image_dataset_from_directory` supported `reshuffle_each_iteration = False`, I could enable this behavior within the call to `image_dataset_from_directory`.  However, since it's not currently supported, I have to manually reconstruct the behavior:
1. Read in the full dataset using `image_dataset_from_directory(..., shuffle=False)`
2. Calculate the size of the dataset: `dataset_size = len(list(dataset))`
    - Somewhat inefficient
3. Call `Dataset.shuffle(..., buffer_size=dataset_size, reshuffle_each_iteration=False)`

I think it would be useful if `image_dataset_from_directory` directly supported this behavior.

**Will this change the current api? How?**

It will add compatibility with `Dataset.shuffle(..., reshuffle_each_iteration=False)` to `image_dataset_from_directory`.

**Who will benefit from this feature?**

Users of `image_dataset_from_directory` who are looking to have repeatability across `Dataset` iterations.

**Comment by rchao** [comment]
@fchollet, wondering if this makes sense to you and should we make this a community contribution welcome item?
